lib/remap-tests/tests/diagnostics/successful_parse_regex_type.vrl

```diff
@@ -0,0 +1,8 @@
+# object: { "message": "bananas and another ant" }
+# result: "anaana"
+
+.result = parse_regex!(.message, /(?P<an>an.)/)
+a = strip_whitespace(.result.an)
+b = strip_whitespace(.result.1)
+
+a + b
```

Comment:
This doesn't belong in "diagnostics", since it isn't testing any diagnostic messages.

I would put it in `tests/examples` (and probably add a description in the comment section explaining what we're actually testing here).